python/inline/conftest.py

```diff
@@ -923,12 +970,19 @@ def run(self, test: InlineTest, out: List) -> None:
         tree = ast.parse(test.to_test())
         codeobj = compile(tree, filename="<ast>", mode="exec")
         start_time = time.time()
-        # TODO: run the test within timeout limit, otherwise raise TimeoutException
-        exec(codeobj, test.globs)
-        end_time = time.time()
-        out.append(f"Test Execution time: {round(end_time - start_time, 4)} seconds")
-        if test.globs:
-            test.globs.clear()
+        if test.timeout >= 0: 
+            with timeout(seconds = test.timeout):
+                exec(codeobj, test.globs)
+                end_time = time.time()
+                out.append(f"Test Execution time: {round(end_time - start_time, 4)} seconds")
+            if test.globs:
+                    test.globs.clear()
+        else: 
+            exec(codeobj, test.globs)
+            end_time = time.time()
+            out.append(f"Test Execution time: {round(end_time - start_time, 4)} seconds")
+            if test.globs:
+                test.globs.clear()
```

Comment:
You can combine the repeated code.
For example
```
if test.timeout >= 0:
    with timeout(seconds = test.timeout):
        exec(codeobj, test.globs)
else:
    exec(codeobj, test.globs)
end_time = time.time()
out.append(f"Test Execution time: {round(end_time - start_time, 4)} seconds")
if test.globs:
    test.globs.clear()
```
Let me know if you have any concerns

Reply:
cleaned up...